packages/tracing/test/hub.test.ts

```diff
@@ -288,15 +344,93 @@ describe('Hub', () => {
         expect(child.sampled).toBe(transaction.sampled);
       });
 
-      it('should propagate sampling decision to child transactions in XHR header', () => {
-        // TODO fix this and write the test
+      it('should propagate positive sampling decision to child transactions in XHR header', () => {
+        const hub = new Hub(
+          new BrowserClient({
+            dsn: 'https://1231@dogs.are.great/1121',
+            tracesSampleRate: 1,
+            integrations: [new BrowserTracing()],
+          }),
+        );
+        jest.spyOn(hubModule, 'getCurrentHub').mockReturnValue(hub);
+
+        const transaction = hub.startTransaction({ name: 'dogpark' });
+        hub.configureScope(scope => {
+          scope.setSpan(transaction);
+        });
+
+        const request = new XMLHttpRequest();
+        request.open('GET', '/chase-partners');
+
+        // mock a response having been received successfully (we have to do it in this roundabout way because readyState
+        // is readonly and changing it doesn't trigger a readystatechange event)
+        Object.defineProperty(request, 'readyState', { value: 4 });
+        request.dispatchEvent(new Event('readystatechange'));
+
+        // this looks weird, it's true, but it's really just `request.impl.flag.requestHeaders` - it's just that the
+        // `impl` key is a symbol rather than a string, and therefore needs to be referred to by reference rather than
+        // value
+        const headers = (request as any)[getSymbolObjectKeyByName(request, 'impl') as symbol].flag.requestHeaders;
```

Comment:
[L] Damn, that's some higher-level magic :D

Reply:
LOL - true. But now I know about symbols!